### PR TITLE
Add category configuring, and default accept unofficial to true for active elements

### LINKED ISSUE #162: Active elements category cannot be changed
The active elements category is hard coded to "active elements". https://github.com/apluslms/a-plus-rst-tools/blob/1a7d6e007f5127e1d68c8ddf0a5fa0cec3845f10/directives/ae_output.py#L109

Fixing this may affect a course if it tries to change the category and nobody noticed that it doesn't work.

### LINKED ISSUE #164: Categories are difficult to configure
Categories cannot be configured easily. E.g. setting `accept_unofficial_submits` is difficult. `append_content` in conf.py can be used but it is a bit cumbersome.

### LINKED ISSUE #165: Default active elements to allow unofficial submissions
This allows submission after the deadline. Needed for https://github.com/apluslms/a-plus/issues/487

#164 should be fixed first so that the behaviour can be easily changed if the teacher wants to.

It should probably not be enabled by default if max submissions is specfied. As `accept_unofficial_submits` is a category specific setting, this would either mean disabling unofficial submission if even one active element exercise has max submissions specified or two different default categories for active elements depending on whether max submission is specified or not.

### PR DESCRIPTION
**Pull request by lainets**
# Description

**What?**

Add easier category configuring, and accept unofficial submissions by default to active elements exercises without max submissions.

**Why?**

To close #165.

**How?**

Active elements with max submissions are put in a separate category "active elements with max submissions" that doesn't accept unofficial submisions by default. 

Closes #162
Closes #164
Closes #165 

# Testing

**What type of test did you run?**

- [ ] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [ ] Django unit tests.
- [ ] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [X] Manual testing.

Tried changing the category options in conf.py, and checked that the changes happen in A+. Also checked that "active elements" category accepts unofficial submission by default and that it can be overridden in conf.py.

### CODE REVIEW

**Comment by lainets** [comment]
Of note is whether the solution to the max submissions/accept unofficial submissions problem is satisfactory. Does changing the category of active elements exercises with max submissions cause issues to some courses?

File: toc_config.py [hunk 1]
```diff
@@ -395,15 +395,31 @@ def parse_chapter(docname, doc, parent, module_meta):
 
     # Create categories.
     category_names = app.config.category_names
+    for key, name in category_names.items():
+        category = app.config.categories.get(key, {"name": name})
+        if name != category.get("name", name):
```
Comment (Mankro) [inline toc_config.py:400]: `category_names` in `conf.py` may set multilingual names. Isn't this comparison a bit suspicious in that case?

```python
category_names = {
    'chapter': {
        'fi': 'Luku',
        'en': 'Chapter',
    },
    'submit': {
        'fi': 'Pisteytetty',
        'en': 'Graded',
    },
    'feedback': {
        'fi': 'Palaute',
        'en': 'Feedback',
    },
}

```

File: toc_config.py [hunk 2]
```diff
@@ -395,15 +395,31 @@ def parse_chapter(docname, doc, parent, module_meta):
...
         key: {
             'name': category_names.get(key, key),
         } for key in category_keys
     }
+
+    for key, category in app.config.categories.items():
+        if key not in categories:
+            raise SphinxError(f"Category with key '{key}' was configured but not used in the material")
+
+        categories[key].update(category)
```
Comment (Mankro) [inline toc_config.py:413]: Should this part be moved after the following `for` loop that forces `chapter` and `feedback` categories to `status` `nototal`? If somebody really wanted to set a different status, then they could do it with the new `categories` setting.

Some courses actually want to show feedback points to the students and changing the status of the feedback category would be then useful.

File: directives/ae_output.py [hunk 3]
```diff
@@ -109,6 +109,8 @@ def run(self):
             'category': 'active elements',
             'max_submissions': self.options.get('submissions', data.get('max_submissions', env.config.ae_default_submissions)),
         })
+        data["category"] = "active elements" if data["max_submissions"] == 0 else "active elements with max submissions"
```
Comment (Mankro) [inline directives/ae_output.py:112]: Looks like it wasn't possible to change the category of the active element in the RST directive options. But we could now add the possibility. If somebody wants to change the category, then we also do not need to set default setting (`accept_unofficial_submits`) to the new category.

Do you think this is OK?

**Review by Mankro — CHANGES_REQUESTED** [review]
Great, this looks good! I have a couple of comments below.

> Of note is whether the solution to the max submissions/accept unofficial submissions problem is satisfactory. Does changing the category of active elements exercises with max submissions cause issues to some courses?

I think it is good. I don't think changing the category hurts any current user.

File: directives/ae_output.py (hunk 3 above)
Comment (lainets) [reply directives/ae_output.py:112]: Yeah, I don't think we want to set `accept_unofficial_submits` on an user specified caregory as we don't really know what the intention is and what else is in that same category. I'll add the possibility of changing the category.

**Review by Mankro — APPROVED** [review]
Excellent!

**Pull request merged**